app/main.py

```diff
@@ -1 +1,21 @@
-# write your code here
+from app.cafe import Cafe
+from app.errors import (
+    NotWearingMaskError,
+    VaccineError
+)
+
+
+def go_to_cafe(friends: list[dict], cafe: "Cafe") -> str:
+    masks_to_buy = 0
+
+    for friend in friends:
+        try:
+            cafe.visit_cafe(friend)
+        except VaccineError:
+            return "All friends should be vaccinated"
```

Comment:
The `VaccineError` exception handling should be moved inside the loop to ensure that each friend's vaccination status is checked independently. This will prevent the loop from exiting prematurely and ensure that all friends are evaluated.